src/provider/controller/NotificationCenter.ts

```diff
@@ -98,6 +90,12 @@ export class NotificationCenter extends AsyncInit {
         }));
     }
 
+    private async onAction(action: RootAction): Promise<void> {
+        if (action instanceof ToggleVisibility) {
+            this.toggleWindow(action.visible || false);
```

Comment:
Don't think this is equivalent to what was here previously. On the action, `visible` is ternary - it can also be `undefined`, at which point it should be the opposite of whatever the current value in the state is.

Though given that these actions now run after the reducer (right?), is it safe to read this value from the store, rather than from the action?..

Reply:
well caught. thanks